core/blocktree/blocktree_test.go

```diff
@@ -159,71 +139,65 @@ func TestNode_isDecendantOf(t *testing.T) {
 }
 
 func TestBlockTree_LongestPath(t *testing.T) {
-	bt := createFlatTree(t, 3)
+	bt, hashes := createFlatTree(t, 3)
 
 	// Insert a block to create a competing path
-	extraBlock := types.BlockWithHash{
-		Header: &types.BlockHeaderWithHash{
-			ParentHash: zeroHash,
+	extraBlock := types.Block{
+		Header: &types.BlockHeader{
+			ParentHash: hashes[0],
 			Number:     big.NewInt(1),
-			Hash:       common.Hash{0xAB},
 		},
 		Body: &types.BlockBody{},
 	}
 
-	bt.AddBlock(extraBlock)
-
-	expectedPath := []*node{
-		bt.GetNode(common.Hash{0x00}),
-		bt.GetNode(common.Hash{0x01}),
-		bt.GetNode(common.Hash{0x02}),
-		bt.GetNode(common.Hash{0x03}),
+	_, err := extraBlock.Header.Hash()
+	if err != nil {
+		t.Fatal(err)
 	}
+	bt.AddBlock(extraBlock)
 
 	longestPath := bt.LongestPath()
 
 	for i, n := range longestPath {
-		if n.hash != expectedPath[i].hash {
-			t.Errorf("expected Hash: 0x%X got: 0x%X\n", expectedPath[i].hash, n.hash)
+		if n.hash != hashes[i] {
+			t.Errorf("expected Hash: 0x%X got: 0x%X\n", hashes[i], n.hash)
 		}
 	}
 }
 
 func TestBlockTree_Subchain(t *testing.T) {
-	bt := createFlatTree(t, 4)
+	bt, hashes := createFlatTree(t, 4)
+	expectedPath := hashes[1:]
 
 	// Insert a block to create a competing path
-	extraBlock := types.BlockWithHash{
-		Header: &types.BlockHeaderWithHash{
-			ParentHash: zeroHash,
+	extraBlock := types.Block{
+		Header: &types.BlockHeader{
+			ParentHash: hashes[0],
 			Number:     big.NewInt(1),
-			Hash:       common.Hash{0xAB},
 		},
 		Body: &types.BlockBody{},
 	}
 
-	bt.AddBlock(extraBlock)
-
-	expectedPath := []*node{
-		bt.GetNode(common.Hash{0x01}),
-		bt.GetNode(common.Hash{0x02}),
-		bt.GetNode(common.Hash{0x03}),
+	_, err := extraBlock.Header.Hash()
+	if err != nil {
+		t.Fatal(err)
 	}
+	bt.AddBlock(extraBlock)
 
-	subChain := bt.SubChain(common.Hash{0x01}, common.Hash{0x03})
+	subChain := bt.SubChain(hashes[1], hashes[3])
 
 	for i, n := range subChain {
-		if n.hash != expectedPath[i].hash {
-			t.Errorf("expected Hash: 0x%X got: 0x%X\n", expectedPath[i].hash, n.hash)
+		if n.hash != expectedPath[i] {
```

Comment:
Should we use `DeepEqual` here and update the name of `expectedPath`?

Reply:
I don't think we can use DeepEqual here, since we're iterating through a list of nodes then comparing node.hash. I think the name is fine, do you have any suggestions?

Reply:
Ah, ok. I think I just misread the first time.